### PR TITLE
[Datafactory] ADLS Gen 2 support for HDI BYOC and vNet support for HDI on demand

### PR DESCRIPTION
**Pull request by hvermis**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json [hunk 1]
```diff
@@ -4406,6 +4410,14 @@
           "items": {
             "$ref": "#/definitions/ScriptAction"
           }
+        },
+        "virtualNetworkId": {
```
Comment (KrisBash) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json]: When adding properties within an API version, there is risk if the properties are updated by a current version client, but then the resource is modified by a down-level client. This is not strictly a breaking change, but you certainly have more control against accidental overwrite if exposed in a new API version. What is the behavior on the service if an update excludes these properties after they have been previously defined?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see questions regarding adding properties within API version

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json (hunk 1 above)
Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json]: We are aware of these scenarios, and our customers also are used to these kinds of updates. We keep adding new features all the time, and they result in new properties, so adding new API version time is not feasible. The service overwrites the entities. This affects our SDK and Rest API users, but they are already aware of this, and keep getting new SDK versions when they want to use new properties. In fact, these changes are to catch up with our UX support.

Comment (KrisBash) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json]: Ok, thanks

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by hvermis** [comment]
@tjprescott Can you please merge this?

**Review by tjprescott — APPROVED** [review]

**Pull request merged**